src/snowflake/snowpark/stored_procedure.py

```diff
@@ -153,6 +154,15 @@ def __call__(
         if self._anonymous_sp_sql:
             call_sql = generate_call_python_sp_sql(session, self.name, *args)
             query = f"{self._anonymous_sp_sql}{call_sql}"
+            if not block:
+                return session._execute_sproc_internal(
+                    query=query,
+                    is_return_table=self._is_return_table,
+                    block=block,
+                    statement_params=statement_params,
+                    ast_stmt=self._ast_stmt,
+                )
+
             if self._is_return_table:
                 qid = session._conn.execute_and_get_sfqid(
                     query, statement_params=statement_params
```

Comment:
is the logic 166-170 encapsulated into `_execute_sproc_internal`? is it possible to do the following?

```python
ret = _execute_sproc_internal(...block=block, return_table=self._is_return_table)
if not block:
    return ret
else:
    ret._ast = sproc_expr
    return ret
```